onchain/example-program-on-bonsol/src/lib.rs

```diff
@@ -0,0 +1,155 @@
+use bonsol_interface::callback::{handle_callback, BonsolCallback};
+use bonsol_interface::instructions::{execute_v1, CallbackConfig, ExecutionConfig, InputRef};
+use solana_program::account_info::AccountInfo;
+use solana_program::entrypoint::ProgramResult;
+use solana_program::program_error::ProgramError;
+use solana_program::pubkey;
+use solana_program::pubkey::Pubkey;
+
+use solana_program::clock::Clock;
+use solana_program::instruction::AccountMeta;
+use solana_program::program::invoke_signed;
+use solana_program::program_memory::sol_memcmp;
+use solana_program::rent::Rent;
+use solana_program::sysvar::Sysvar;
+use solana_program::{declare_id, entrypoint, msg, system_instruction};
+use std::str::from_utf8;
+
+declare_id!("exay1T7QqsJPNcwzMiWubR6vZnqrgM16jZRraHgqBGG");
+const SIMPLE_IMAGE_ID: &str = "68f4b0c5f9ce034aa60ceb264a18d6c410a3af68fafd931bcfd9ebe7c1e42960";
+
+static EA1: Pubkey = pubkey!("3b6DR2gbTJwrrX27VLEZ2FJcHrDvTSLKEcTLVhdxCoaf");
+static EA2: Pubkey = pubkey!("g7dD1FHSemkUQrX1Eak37wzvDjscgBW2pFCENwjLdMX");
+static EA3: Pubkey = pubkey!("FHab8zDcP1DooZqXHWQowikqtXJb1eNHc46FEh1KejmX");
+
+entrypoint!(main);
+fn main<'a>(
+    _program_id: &Pubkey,
+    accounts: &'a [AccountInfo<'a>],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    let (ix, data) = instruction_data.split_at(1);
```

Comment:
I'd consider checking **data** is at least 57 bytes before all slicing below.

On the subject the code below, lines 35-40 are a bit dense to read.  What do you think of making that a method in itself, or separating operations onto different lines for legibility?